pkg/apis/kubernetes/client.go

```diff
@@ -52,68 +63,123 @@ func init() {
 }
 
 // Kube returns the clientset for Kubernetes upstream.
-func (c *K8sClient) KubeInterface() kubernetes.Interface {
+func (c *CustomClient) KubeInterface() kubernetes.Interface {
 	return c.kubeClient
 }
 
 // Return a rest client to perform CRUD operations on Kubernetes objects
-func (c *K8sClient) KubeRest() crclient.Client {
+func (c *CustomClient) KubeRest() crclient.Client {
 	return c.crClient
 }
 
-func (c *K8sClient) PipelineClient() pipelineclientset.Interface {
+func (c *CustomClient) PipelineClient() pipelineclientset.Interface {
 	return c.pipelineClient
 }
 
-func (c *K8sClient) JvmbuildserviceClient() jvmbuildserviceclientset.Interface {
+func (c *CustomClient) JvmbuildserviceClient() jvmbuildserviceclientset.Interface {
 	return c.jvmbuildserviceClient
 }
 
 // Returns a DynamicClient interface.
 // Note: other client interfaces are likely preferred, except in rare cases.
-func (c *K8sClient) DynamicClient() dynamic.Interface {
+func (c *CustomClient) DynamicClient() dynamic.Interface {
 	return c.dynamicClient
 }
 
-// NewHASClient creates kubernetes client wrapper
-func NewK8SClient() (*K8sClient, error) {
+// NewK8SClients creates kubernetes client wrapper
+func NewK8SClients() (*K8sClients, error) {
+	// Init (workload) cluster client
+	clusterKubeconfigPath := os.Getenv("CLUSTER_KUBECONFIG")
```

Comment:
What if the env is not defined?. Should not exit with an descriptive error?

Reply:
yeah good point, I plan to update other errors in controller & kubeclients' init functions too, so they are more descriptive. But first I need a decision from the teams whether we go with this approach or not.

Reply:
updated